src/services/completions.ts

```diff
@@ -880,6 +889,22 @@ function completionInfoFromData(
         }
     }
 
+    // When the completion is for the expression of a case clause (e.g. `case |`),
+    // filter literals & enum symbols whose values are already present in existing case clauses.
+    let caseClause = findAncestor(contextToken, isCaseClause);
+    if (caseClause && (isCaseKeyword(contextToken!) || isNodeDescendantOf(contextToken!, caseClause.expression))) {
+        const tracker = newCaseClauseTracker(checker, caseClause.parent.clauses);
+        literals = literals.filter(literal => !tracker.hasValue(literal));
+        symbols.forEach((symbol, i) => {
+            if (symbol.valueDeclaration && isEnumMember(symbol.valueDeclaration)) {

```

Comment:
Note that the filtering won't work if someone is using e.g. an object literal as an enum, because `checker.constantValue` only works for enum members. But I'm hoping that filtering out literals and enum members already covers a lot of the cases.

Reply:
Seems like something that can be expanded to `readonly` members with literal types if there’s demand for it 👍